### PR TITLE
Relax github transform types

### PR DESCRIPTION
**Pull request by jgrigg**
## Hey, I just made a Pull Request!

Sorry couldn't seem to reopen #18735 cc 

@benjdlambert I've resolved [this](https://github.com/backstage/backstage/pull/18735#pullrequestreview-1552312024) i think

At seek we would like to represent github teams as resources (ACLs) rather than groups. This change relaxes the type on the transform functions to allow them to return any `Entity`.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jgrigg** [comment]
Oh god new mac (M2) and have run into the `canvas` issue 🤦 . Need to figure out how to Rosetta my way out of hell.

**Comment by benjdlambert** [comment]
@jgrigg Looks like theres some test failures here too that need updating because of the output from some of these private functions with teams etc.

**Review by gstamp — APPROVED** [review]

**Comment by jgrigg** [comment]
> @jgrigg Looks like theres some test failures here too that need updating because of the output from some of these private functions with teams etc.

All sorted :)

**Comment by benjdlambert** [comment]
@backstage/catalog-maintainers for review here too

File: packages/catalog-model/api-report.md [hunk 1]
```diff
@@ -53,6 +53,16 @@ export { ApiEntityV1alpha1 };
 // @public
 export const apiEntityV1alpha1Validator: KindValidator;
 
+// @public (undocumented)
+export function areGroupEntities(
+  entities: Entity[],
+): entities is GroupEntityV1alpha1[];
```
Comment (freben) [inline packages/catalog-model/api-report.md:59]: I'd prefer not to have these additions to the API actually. We are still figuring out what the best way forward is for proper usage of apiVersions. If we settle on a v1 of Group we'd be tempted to change the return type of these to that "new most wanted" version of a group potentially leading to breakages. Plus arguably the contract of the functions (acting on lists) is a bit oddly specific from an outsider's point of view.

Maybe just inline these privately where they are used?

Comment (freben) [reply packages/catalog-model/api-report.md:59]: Oh haha, I wrote this before realizing we already have the one-entity version of the same already :) Well, I think the comment still stands, let's not dig this particular hole any deeper.

Comment (jgrigg) [reply packages/catalog-model/api-report.md:59]: cool, I'll do this locally to the plugin 👍

Comment (jgrigg) [reply packages/catalog-model/api-report.md]: done :)

File: .changeset/young-ants-nail.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-catalog-backend-module-github': patch
+'@backstage/catalog-model': patch
```
Comment (benjdlambert) [inline .changeset/young-ants-nail.md:3]: Don't think we need this here anymore as we're not touching this package anymore 👍

Comment (jgrigg) [reply .changeset/young-ants-nail.md]: good call 👍

**Review by freben — APPROVED** [review]
Looks good to me!

**Pull request merged**